### PR TITLE
remove refs to sqlite db if disabled in CMake

### PR DESCRIPTION
**Pull request by JehandadKhan**
If the SQLite machinery is disabled in CMake then MIOpen should build without it. This PR addresses spurious refs to SQLite files and the respective machinery. This PR also formalizes the dependence of the Binary Cache on SQLite.

### CODE REVIEW

**Comment by daniellowell** [comment]
Tidy errors.

**Comment by daniellowell** [comment]
Tidy issues:

```
[ 82%] Linking CXX executable ../bin/test_cache

CMakeFiles/test_cache.dir/cache.cpp.o: In function `check_bz2_compress()':

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0x10e): undefined reference to `miopen::compress(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool*)'

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0x400): undefined reference to `miopen::compress(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool*)'

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0x583): undefined reference to `miopen::compress(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool*)'

CMakeFiles/test_cache.dir/cache.cpp.o: In function `check_bz2_decompress()':

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0x820): undefined reference to `miopen::decompress(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int)'

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0xa93): undefined reference to `miopen::compress(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool*)'

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0xc40): undefined reference to `miopen::decompress(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int)'

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0xe09): undefined reference to `miopen::decompress(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int)'

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0xfba): undefined reference to `miopen::decompress(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int)'

CMakeFiles/test_cache.dir/cache.cpp.o: In function `check_kern_db()':

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0x15a9): undefined reference to `miopen::KernDb::KernDb(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long)'

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0x1776): undefined reference to `miopen::KernDb::KernDb(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long)'

/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0x1acc): undefined reference to `miopen::KernDb::KernDb(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long, std::function<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool*)>, std::function<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int)>)'

clang: error: linker command failed with exit code 1 (use -v to see invocation)

test/CMakeFiles/test_cache.dir/build.make:99: recipe for target 'bin/test_cache' failed
```

**Comment by daniellowell** [comment]
@JehandadKhan Status?

**Comment by daniellowell** [comment]
@JehandadKhan I think I introduced a missing definition with my merge with develop. The test_cache test doesn't compile since it is missing:
```
/var/jenkins/workspace/s_MIOpen_sqlite_optional_compile/test/cache.cpp:(.text+0x21bd): undefined reference to `miopen::GetCacheFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'

CMakeFiles/test_cache.dir/cache.cpp.o: In function `check_cache_str()':
```

I'm backing away now so I don't do more damage.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -124,8 +124,6 @@ set( MIOpen_Source
     dropout_api.cpp
     readonlyramdb.cpp
     execution_context.cpp
-    kern_db.cpp
-    bz2.cpp
```
Comment (daniellowell) [inline src/CMakeLists.txt:128]: Why do these go away?

Comment (JehandadKhan) [reply src/CMakeLists.txt:128]: Only added if SQLite is enabled.

**Pull request merged**